deb_helper/README.md

```diff
@@ -0,0 +1,192 @@
+        "libpython2.7-minimal": "pool/main/p/python2.7/libpython2.7-minimal_2.7.9-2+deb8u1_amd64.deb",
+        "libpython2.7-stdlib": "pool/main/p/python2.7/libpython2.7-stdlib_2.7.9-2+deb8u1_amd64.deb",
+        "python2.7-minimal": "pool/main/p/python2.7/python2.7-minimal_2.7.9-2+deb8u1_amd64.deb",
+        "zlib1g": "pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_amd64.deb",
+    },
+    packages_sha256 = {
+        "libpython2.7-minimal": "916e2c541aa954239cb8da45d1d7e4ecec232b24d3af8982e76bf43d3e1758f3",
+        "libpython2.7-stdlib": "cf1c9dfc12d6cfd42bb14bfb46ee3cec0f6ebc720a1419f017396739953b12c5",
+        "python2.7-minimal": "c89199f908d5a508d8d404efc0e1aef3d9db59ea23bd4532df9e59941643fcfb",
+        "zlib1g": "b75102f61ace79c14ea6f06fdd9509825ee2af694c6aa503253df4e6659d6772",
+    },
+)
+```
+
+Internally `.deb` files referenced here will be downloaded by the `deb_loader` tool, renamed to their SHA256 hash (not all characters used in file names are legal in bazel names) and made available in a dictionary named the same as the `deb_packages` rule.
+This dictionary is made available in a file named `deb_packages.bzl` in the `debs` subfolder of this rule.
+You can find the generated and downloaded files in the `./bazel-distroless/external/your_rule_name/debs` folder after building the project if you're interested.
+
+To actually use the `.deb` files in a BUILD file rule like `docker_build`, you first have to load all dictionaries of package sources you want to use.
+This is done with the `load("@your_rule_name//debs:deb_packages.bzl", "your_rule_name")` line.
+Then you can use the dictionary named the same as the `deb_packages` rule to refer to the packages you defined in the WORKSPACE file.
+
+```python
+load("@debian_jessie_amd64//debs:deb_packages.bzl", "debian_jessie_amd64")
+
+docker_build(
+    name = "python27",
+    base = "//base:base",
+    debs = [
+        debian_jessie_amd64["libpython2.7-minimal"],
+        debian_jessie_amd64["libpython2.7-stdlib"],
+        debian_jessie_amd64["python2.7-minimal"],
+        debian_jessie_amd64["zlib1g"],
+        ],
+    entrypoint = [
+        "/usr/bin/python2.7",
+    ],
+    symlinks = {
+        "/usr/bin/python": "/usr/bin/python2.7",
+    },
+)
+```
+
+### **Using the `update_workspace` helper program**
+
+As you might have noticed, it is a lot of work to get the exact path and SHA256 hash of a `.deb` package.
+All this information is already available online at the mirrors you defined in your WORKSPACE file: As soon as you know the exact distribution, package name and version, you should be able to just generate this data on the fly.
+
+`update_workspace` is exactly such a helper program.
+It uses the `buildifier` and `buildozer` tools from [https://github.com/bazelbuild/buildtools](https://github.com/bazelbuild/buildtools), which need to be available on your $PATH.
+
+To update hashes and paths in a single `deb_packages` rule with the `update_workspace --update/-u RULENAME` switch.
```

Comment:
True, seems reasonable and better fitting to make everything available to a `bazel` invocation.

Reply:
Looked deeper into it and I think it might be reasonable to just always run the `--add` functionality when invoked that way which adds missing packages and updates the paths/hashes. It also could remove packages that are not referred in docker_build rules, but I guess there might be other use cases for having a *.deb file available in your build environment, such as the cacerts example.

Reply:
Revisiting this one... it is surprisingly hard to package Python executables that depend on external packages. Python-apt does not make the job easier at all (they don't publish to pypi, contain C bindings and of course are not using Bazel to build).

I'll probably need to write a BUILD file for the project there and then also stop using subpar who don't support C dependencies yet. In theory the approach from gazelle works (and I have it running the binary already, it'll just fail as soon as it is supposed to parse a file), but to get it done with Python is tougher than expected.